sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/DistinctMapTest.java

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.implementation.query.DistinctMap;
+import com.azure.cosmos.implementation.query.DistinctQueryType;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class DistinctMapTest {
+    @DataProvider(name = "distinctMapArgProvider")
+    public Object[][] distinctMapArgProvider() {
+        return new Object[][] {
+            {DistinctQueryType.Ordered},
+            {DistinctQueryType.Unordered},
+        };
+    }
+
+    @Test(groups = "unit", dataProvider = "distinctMapArgProvider")
+    public void integerValue(DistinctQueryType queryType) {
+        String resource = String.format("{ " + "\"id\": %d + \"}", 5);
+        DistinctMap distinctMap = DistinctMap.create(queryType, null);
+        Utils.ValueHolder<String> outHash = new Utils.ValueHolder<>();
+        boolean add = distinctMap.add(resource, outHash);
+        System.out.println("outHash5 = " + outHash.v);
+        assertThat(add).as("Value should be added first time").isTrue();
+        boolean add2 = distinctMap.add(resource, outHash);
+        assertThat(add2).as("same value should not be added again").isFalse();
+        resource = String.format("{ " + "\"id\": %d + \"}", 3);
+        boolean add3 = distinctMap.add(resource, outHash);
+        assertThat(add3).as("different value should be added again").isTrue();
+    }
+
+    @Test(groups = "unit", dataProvider = "distinctMapArgProvider")
+    public void stringValue(DistinctQueryType queryType) {
+        String resource = String.format("{ " + "\"id\": \"5\" + \"}");
+        DistinctMap distinctMap = DistinctMap.create(queryType, null);
+
+        Utils.ValueHolder<String> outHash = new Utils.ValueHolder<>();
+        boolean add = distinctMap.add(resource, outHash);
+        assertThat(add).as("Value should be added first time").isTrue();
+        boolean add2 = distinctMap.add(resource, outHash);
+        assertThat(add2).as("same value should not be added again").isFalse();
+        resource = String.format("{ " + "\"id\": \"6\" + \"}");
+        boolean add3 = distinctMap.add(resource, outHash);
+        assertThat(add3).as("different value should be added again").isTrue();
+    }
+
+    @Test(groups = "unit", dataProvider = "distinctMapArgProvider")
+    public void objectValue(DistinctQueryType queryType) {
```

Comment:
We need a test for two objects where the property values are just shuffeled:

```
{
     "name" : "John",
     "age": 24
}
```

vs 

```
{
     "age": 24,
     "name" : "John"
}
```

And they should hash to the same, since ordered of JSON properties does not matter.

Reply:
I see that you have one.

Reply:
Yeah added one for this